Removed select star and group by

nadeeshaan opened the pull request:
Removed select\* and group by and instead added the column names and select distinct

nadeeshaan committed:
Removed select star and group by

nadeeshaan commented:
Testing:
Query can be found - `JROOT/administrator/components/com_menus/models/menus.php`
In order to check the query execution log in to the admin console and click menu manager. Then the query triggers.

Change in the Query: In this query group by clause has been removed and instead added the select distinct. Also removed the select \* and instead added the corresponding fields

nadeeshaan committed:
removed distinct and added where clause

gunjanpatel commented on administrator/components/com_menus/models/menus.php:
```diff
@@ -172,9 +171,6 @@ protected function getListQuery()
 		}
 
-		// Add the list ordering clause.
-		$query->order($db->escape($this->getState('list.ordering', 'a.id')) . ' ' . $db->escape($this->getState('list.direction', 'ASC')));
```
You are removing Ordering? Any reason?

alikon replied:
we need to split this query
to allow the query work when you click on order by title

gunjanpatel replied:
@alikon What is the reason behind to not ordering by `a.id` specifically? I mean if you are thinking to remove it then it will make an issue in UX. 
![menu manager menus](https://cloud.githubusercontent.com/assets/2002921/3800803/d1e68fbe-1bfc-11e4-84e2-9276f1f8c922.png)

gunjanpatel replied:
According to discussion on slack @nadeeshaan  please roll back ordering. 
Thanks.

nadeeshaan committed:
roll back order by clause

nadeeshaan commented:
## Testing Instructions:

In order to check the query execution log in to the admin console and click menu manager. Then the query triggers. To check how the PR works, first apply the PR via the Joomla! Patch Tester and then log in to the admin console and click menu manager.

![Menu Manager](https://cloud.githubusercontent.com/assets/1329674/3903334/9b212870-22ca-11e4-86d5-29598a04fe46.png)

### Original Query before the Optimization

``` sql
    SELECT a.*
    FROM #_menu_types AS a
    GROUP BY a.id, a.menutype, a.title, a.description
    ORDER BY a.id asc
```

### After the Optimization

``` sql
    SELECT a.id, a.menutype, a.title, a.description
    FROM #_menu_types AS a
    WHERE a.id > 0
    ORDER BY a.id asc
```

### Query Change Description

In this query group by clause has been removed. Also removed the `SELECT a.*` and instead added the corresponding fields to the select clause. Also added the `WHERE` clause in order to be able to use the `PRIMARY KEY` as the index.

gunjanpatel commented:
@nadeeshaan I am not able to find PR for this in Joomla-cms. Can you please point it out?

orware commented:
Taking a look at the current code:
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_menus/models/menus.php#L164

It appears this one has already been integrated into the main Joomla CMS code base so we can close this pull request here in this repo to prevent confusion ;-).

gunjanpatel commented:
Yes, indeed. Here is a PR https://github.com/joomla/joomla-cms/pull/4112
